src/apps_plugin.c

```diff
@@ -725,18 +763,24 @@ long all_pids_count = 0;
 
 struct pid_stat *get_pid_entry(pid_t pid)
 {
+	// FIXME: The double linked list pointed by root_of_pids
+	//        is garanteed to be a sorted list?!
+
 	if(all_pids[pid]) {
 		all_pids[pid]->new_entry = 0;
 		return all_pids[pid];
 	}
 
-	all_pids[pid] = calloc(sizeof(struct pid_stat), 1);
+	// FIX: Just to use the 'correct' semantics.
+	all_pids[pid] = calloc(1, sizeof(struct pid_stat));
 	if(!all_pids[pid]) {
 		error("Cannot allocate %lu bytes of memory", (unsigned long)sizeof(struct pid_stat));
 		return NULL;
 	}
 
-	all_pids[pid]->fds = calloc(sizeof(int), 100);
+	// FIXME: Why the arbitrary value 100 was choosen for the maximum number of descriptors?
```

Comment:
It is just a value to start with. Sure we are wasting some memory with this.
We should probably #define it to 10 or something.